src/main/java/com/forum/forum_backend/services/PostServiceImpl.java

```diff
@@ -0,0 +1,181 @@
+package com.forum.forum_backend.services;
+
+import com.forum.forum_backend.config.UserPrincipal;
+import com.forum.forum_backend.dtos.PaginatedResponse;
+import com.forum.forum_backend.dtos.PostDto;
+import com.forum.forum_backend.dtos.UserDto;
+import com.forum.forum_backend.exceptions.NotFoundException;
+import com.forum.forum_backend.exceptions.UnauthorizedException;
+import com.forum.forum_backend.models.PostEntity;
+import com.forum.forum_backend.models.ThreadEntity;
+import com.forum.forum_backend.models.UserEntity;
+import com.forum.forum_backend.repositories.PostRepository;
+import com.forum.forum_backend.repositories.ThreadRepository;
+import com.forum.forum_backend.repositories.UserRepository;
+import com.forum.forum_backend.services.interfaces.PostService;
+import com.forum.forum_backend.services.interfaces.UserService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityNotFoundException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+@Transactional
+@Service
+public class PostServiceImpl implements PostService {
+
+	private final PostRepository postRepository;
+	private final ThreadRepository threadRepository;
+	private final UserService userService;
+	private final UserRepository userRepository;
+
+	public PostServiceImpl(PostRepository postRepository, ThreadRepository threadRepository, UserService userService, UserRepository userRepository) {
+		this.postRepository = postRepository;
+		this.threadRepository = threadRepository;
+		this.userService = userService;
+		this.userRepository = userRepository;
+	}
+
+	@Override
+	public PostDto addPost( int threadId, PostDto postDto) throws NotFoundException, UnauthorizedException {
+		try {
+			ThreadEntity thread = threadRepository.getOne(threadId);
+
+			if (thread.isLocked()) {
+				throw new UnauthorizedException("Thread with id = " + threadId + " is locked, cannot add post");
```

Comment:
zamiast 403, powinno być raczej 400 lub 500, bo 403 to tak jakby ktoś mógł dodawać posty, kiedy temat jest zamknięty.

Reply:
dodałem BadRequestException z status code 400, już podmienione